eclipse-che/che-operator
# feat: accessible workspace healthz endpoint

### What does this PR do?
Makes `/healthz` path of main workspace endpoint accessible from the outside. DevWorkspace Operator needs this to check whether workspace is alive.

TODO:

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20481


### How to test this PR?

latest image here --> `quay.io/mvala/che-operator:gh20481-workspaceHealthzEndpoint`

1. deploy with devworkspaces on openshift
2. start any workspace
3. copy direct workspace url `kubectl get dw -n <user-namespace>`
4. append `/healthz` and try to curl it. You should get 404 response. That is ok, because Theia does not actually implement that endpoint. In case this PR is not working, you would be redirected to openshift login page.
5. try to curl any other workspace url, you should be redirected to openshift login page. Only `<main-url>/healthz` must be accessible.

```bash
chectl server:deploy \
     --installer operator \
     --platform openshift \
     --che-operator-image quay.io/mvala/che-operator:gh20481-workspaceHealthzEndpoint \
     --workspace-engine=dev-workspace
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
